File: be/sensorReadingApi/src/main/java/com/abdo/sensorReadingApi/service/DeviceReadingService.java
package com.abdo.sensorReadingApi.service;

import com.abdo.sensorReadingApi.dto.request.DeviceReadingRequest;
import com.abdo.sensorReadingApi.dto.response.DeviceReadingResponse;

public interface DeviceReadingService {
    Boolean addDeviceReading(DeviceReadingRequest deviceReadingRequest);
    DeviceReadingResponse getLatestReadingList();

    DeviceReadingResponse getDeviceReadingList(Long id);
}

File: fe/sensorReadingUi/src/app/service/deviceReadingService.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({providedIn: 'root'})
export class DeviceReadingService {
  private getLatestReadingsUrl = 'http://localhost:8080/deviceReading/list';
  private addDeviceReadingUrl='http://localhost:8080/deviceReading/add';
  private getDeviceReadingListUrl='http://localhost:8080/deviceReading/details/'
  

  constructor(private http: HttpClient) {}

  getLatestReadingList() {
    return this.http.get(this.getLatestReadingsUrl);
  }

  addDeviceReading(data:any) {
    this.http.post(this.addDeviceReadingUrl, data).subscribe(response => {
    })  ;
  }

  getDeviceReadingList(deviceId:string) {
    return this.http.get(this.getDeviceReadingListUrl+deviceId);
  }
}